building-and-packaging/material/systempaths-and-librarytools_slides.md

```diff
@@ -0,0 +1,289 @@
+    - Shared code must be installed and accessible
+    - Executables can share library in memory
+
+---
+
+## Filesystem Hierarchy Standard
+
+- Defines filesystem layout and location of common files on Linux
+
+<img src="https://raw.githubusercontent.com/Simulation-Software-Engineering/Lecture-Material/add-overview-systempaths-librarytools/building-and-packaging/material/figs/filesystem_hierarchy_structure/fig.png" width=100%; style="margin-left:auto; margin-right:auto; padding-top: 25px; padding-bottom: 25px">
+
+- [Official homepage of FHS](https://refspecs.linuxfoundation.org/fhs.shtml)
+- Certain structure should
+
+**TODO** Set this link to the GitHub main branch before merging.
+
+---
+
+## Some Environment Variables
+
+- `HOME` or `~`: User's home directory
+- `PWD`: Current working directory
+- `PATH`: Location of executable
+- `LC_<NAME>`: Locale's
+- `LANG`: Systems' language
+- System variables often set from `/etc/profile` and `/etc/bashrc` or `/etc/bash.bashrc`
+
+---
+
+## Manipulating Environment Variables 1/2
+
+- Show content of paths via
+
+  ```bash
+  echo $VARIABLENAME
+  printenv VARIABLENAME
+  ```
+
+- Setting variable for a single command (bash)
+
+  ```bash
+  ENVVARIABLE=VALUE ./COMMAND
+  ```
+
+- Setting **environment** variable
+
+  ```bash
+  export ENVVARIABLE=VALUE
+  ```
+
+---
+
+## Manipulating Environment Variables 2/2
+
+- Adding additional values by colon (`:`) separated list
+
+  ```bash
+  PATH=ADDITONALPATH:${PATH}
+  ```
+
+  ```bash
+  PATH=/home/jaustar/bin:${PATH}
+  ```
+
+- Extra paths of user often set in `${HOME}/.bashrc`, `${HOME}/.bash_exports`, `${HOME}/.bash_profile`,...
+
+---
+
+## Some Environment Variables (C++)
+
+- `LD_<NAME>`: Loader related variable
+    - `LD_LIBRARY_PATH`: Library paths for runtime
+- `LIBRARY_PATH`: Library path for compilation (linker)
+- `CPATH`, `C_INCLUDE_PATH`, `CPLUS_INCLUDE_PATH`: Include file path
+- Names can vary beween OS, software.
+- Variables should be manipulated carefully.
+
+---
+
+## Paths and Environment Variables (Python)
+
+- `/usr/`: System-wide installation
+    - `/usr/bin`: Executable scripts
+    - `/usr/lib/pythonX.Y`: Modules/packages
+- `${HOME}/.local`: As above
+    - `${HOME}/.local/bin` and `${HOME}/.local/lib/pythonX.Y`
+- `PYTHONPATH`: Environment variable for non-standard installation of modules/packages
+    - Points to `prefix/lib/pythonX.Y/site-packages`
+
+---
+
+## Working with Libraries
+
+- How to find libraries?
+    - Paths?
+- How to incorporate libraries when compiling?
+    - Compileflags?
+
+---
+
+## Tools for Finding Libraries and Packages
+
+- [ldconfig](https://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html)
+    - Managing paths of library loader
+- [pkg-config]((https://www.freedesktop.org/wiki/Software/pkg-config/))
+    - Manage package information such as
+        - Version
+        - Metadata
+        - Compile and link flags
+
+
+---
+
+## ldconfig
+
+- Generate links and cache to libraries
```

Comment:
I don't understand the "and cache" part here

Reply:
I will make this more verbose. `ldconfig` maintains a cache of all (known) libraries installed on the system which is used by the [library loader/runtime linker](https://man7.org/linux/man-pages/man8/ldconfig.8.html).